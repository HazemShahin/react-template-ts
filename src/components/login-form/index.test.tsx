import * as React from 'react';
import { shallow } from 'enzyme';

import { LoginForm } from '.';

jest.mock('react-redux', () => ({
	connect: (): jest.Mock => jest.fn()
}));

jest.mock('@src/utilities/hooks', () => ({
	useAppSelector: jest.fn(() => ({
		loading: false,
		loginError: ''
	}))
}));

describe('LoginForm component', () => {
	it('should render successfully', () => {
		const tree = shallow(<LoginForm onSubmit={(): jest.Mock => jest.fn()} />);

		expect(tree).toMatchSnapshot();
	});
});
