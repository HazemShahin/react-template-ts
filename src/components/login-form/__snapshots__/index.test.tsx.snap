// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component should render successfully 1`] = `
<form
  className="c-form c-form--login"
  onSubmit={[Function]}
>
  <h2>
    Login
  </h2>
  <Component
    label="Email Address"
    name="email"
    placeholder="someone@example.com"
    register={[Function]}
    type="email"
  />
  <Component
    label="Password"
    name="password"
    placeholder="********"
    register={[Function]}
    type="password"
  />
  <Component
    className=""
    disabled={false}
    type="submit"
  >
    Login
  </Component>
</form>
`;
